services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/diagnostics/growthResults.tsx

```diff
@@ -0,0 +1,160 @@
+    const postPercentageProficient = 100 - Math.round(postPercentageNotProficient)
+    const numberOfStudentsNeedingPracticeInPre = not_yet_proficient_in_pre_test_student_names.length
+    const prePercentageNotProficient = (numberOfStudentsNeedingPracticeInPre/completedStudentCount) * 100
+    const prePercentageProficient = 100 - Math.round(prePercentageNotProficient)
+    const delta = prePercentageProficient ? ((postPercentageProficient - prePercentageProficient) / prePercentageProficient) * 100 : postPercentageProficient
+
+    let needPracticeElement = <span className="need-practice-element no-practice-needed">No practice needed</span>
+    let growthElement = delta > 0 ? <span className="growth-element">{triangleUpIcon}<span>{delta}% growth</span></span> : <span className="growth-element no-growth">No growth</span>
+
+    if (numberOfStudentsNeedingPracticeInPost) {
+      const tooltipText = `<p>${not_yet_proficient_in_post_test_student_names.join('<br>')}</p>`
+      const tooltipTriggerText = numberOfStudentsNeedingPracticeInPost === 1 ? "1 student needs practice" : `${numberOfStudentsNeedingPracticeInPost} students need practice`
+      needPracticeElement = (<Tooltip
+        tooltipText={tooltipText}
+        tooltipTriggerText={tooltipTriggerText}
+        tooltipTriggerTextClass="need-practice-element"
+      />)
+    }
+
+    cardContent = (<React.Fragment>
+      <div className="percentage-circles">
+        <div>
+          <span className="percentage-circle-label">Pre proficient</span>
+          <PercentageCircle
+            bgcolor="#ebebeb"
+            borderWidth={8}
+            color="#9e9e9e"
+            innerColor="#ffffff"
+            percent={prePercentageProficient}
+            radius={52}
+          />
+        </div>
+        <div>
+          <span className="percentage-circle-label">Post proficient</span>
+          <PercentageCircle
+            bgcolor="#ebebeb"
+            borderWidth={8}
+            color="#4ea500"
+            innerColor="#ffffff"
+            percent={postPercentageProficient}
+            radius={52}
+          />
+        </div>
+      </div>
+      <div className="card-footer">
+        {needPracticeElement}
+        {growthElement}
+      </div>
+    </React.Fragment>)
+  }
+
+  return (<section className="skill-group-summary-card">
+    <div className="card-header">
+      <span className="skill-group-name">{name}</span>
+      <SkillGroupTooltip description={description} name={name} />
+    </div>
+    {cardContent}
+  </section>)
+}
+
+const GrowthResults = ({ passedStudentResults, passedSkillGroupSummaries, match, mobileNavigation, }) => {
+  const [loading, setLoading] = React.useState<boolean>(!passedStudentResults);
+  const [studentResults, setStudentResults] = React.useState(passedStudentResults || []);
+  const [skillGroupSummaries, setSkillGroupSummaries] = React.useState(passedSkillGroupSummaries || []);
+  const [openPopover, setOpenPopover] = React.useState({})
```

Comment:
would be nice to have interface defs for these pieces of state

Reply:
you're right - I've already made some edits to this component but I'll go through and add them in the PR that's branched off of this.